### ISSUE TITLE
Deploying both Colonist Landers in the same turn always causes a Game Over

### ISSUE THREAD

**Issue by belgianguy** [OutpostUniverse/OPHD#792]
It seems after landing the Colonist landers the population size callback is fired/processed too late (it should increase the population depending on your currently landed colonist landers). 

In MapViewStateTurn.cpp, line 534

Currently, after the colony landers have landed, the game logic evaluates the population size (which is still 0 at that point) and also evaluates the number of colonist landers still in orbit (also 0 at that point). This condition leads to an immedeate game over.

```
	// Check for Game Over conditions
	if (mPopulation.size() < 1 && mLandersColonist == 0)
	{
		hideUi();
		mGameOverDialog.show();
	}
```

**Comment by ldicker83** [comment]
Good find. Will investigate this since the lander deployment should happen before this check.

**Comment by ldicker83** [comment]
So I'm curious about the setup that you have when dropping both colonist landers? Because it'll lead to an instant game over for sure in two cases: Command Center isn't built and Cargo Landers haven't been landed yet (no food).

Will continue to investigate because this may be related to a similar issue I fixed in the last tagged branch.

**Comment by belgianguy** [comment]
Ah, I see. The Command Center was built, but then I called in both the Cargo Landers and the Colonist Landers down in the same turn. I guess that causes them to die of starvation.

Well, maybe I need training wheels or a Beginner mode, which could warn me of such folly. 

I also tried bringing down the Cargo Landers first, then it indeed did not lead to a game over.

**Comment by ldicker83** [comment]
Yeah, I'm thinking I should probably have that either prioritize the cargo landers so that food is moved to command center storage _before_ population food consumption is processed or just not allow colonist landers to land before cargo landers are dropped.

Also agreed -- a beginner mode is something definitely planned. Between GUI quality of life improvements that I want to implement and providing easier gameplay modes it will let newer users have an easier time with it. My strategy for development is to test it with 'hard mode' sort of being the default and then making it easier versus starting it a lot easier and then making it hard. A case of balancing it so that hard mode isn't impossible to play.

Regardless, thanks for the report! It helps me to know where to focus my efforts especially with documentation (both in the code and for the average user)!

**Comment by Kyrros** [comment]
Little late to the party on this discussion - but I'd lean towards having the item counts resolve themselves first instead of just blocking actions on early turns - as you could run into the issue again on later turns if your food intake/productions match while having zero stored or if you have dangerously low colonist counts when trying to resolve births/deaths.

**Comment by ldicker83** [comment]
Good point. Though this really is fairly isolated to the cargo landers and colonist landers. A fix should be pretty straight forward, just make sure the cargo landers deploy and move their loads before the colonist landers get updated. Should solve that fairly easily.

**Comment by belgianguy** [comment]
In MapViewStateTurn.cpp, I noticed the order of the following functions seem to trigger the death of my colony.
(I always wait until my Command Center is completely built).


```
void MapViewState::nextTurn()
{
        ...

	updatePopulation();
	countFood();
        
        ...
}
```

And when I replaced it by the below order:

```
void MapViewState::nextTurn()
{
        ...

	countFood();
	updatePopulation();
        
        ...
}
```

It no longer occurred (given that the Command Center is fully built).
If the Colony Landers and Cargo Landers are brought down before the Command Center is built, it still triggers a Game Over, which I think is correct.
